tests/e2e/p/interchain_workflow.go

```diff
@@ -0,0 +1,220 @@
+			minStakeDuration := cast.ToDuration(network.GetConfig().DefaultFlags[config.MinStakeDurationKey])
+			require.Equal(testnet.DefaultMinStakeDuration, minStakeDuration)
+		})
+
+		ginkgo.By("creating wallet with a funded key to send from and recipient key to deliver to")
+		factory := secp256k1.Factory{}
+		recipientKey, err := factory.NewPrivateKey()
+		require.NoError(err)
+		keychain := e2e.Env.NewKeychain(1)
+		keychain.Add(recipientKey)
+		nodeURI := e2e.Env.GetRandomNodeURI()
+		baseWallet := e2e.Env.NewWallet(keychain, nodeURI)
+		xWallet := baseWallet.X()
+		cWallet := baseWallet.C()
+		pWallet := baseWallet.P()
+
+		ginkgo.By("defining common configuration")
+		recipientEthAddress := evm.GetEthAddress(recipientKey)
+		avaxAssetID := xWallet.AVAXAssetID()
+		// Use the same owner for sending to X-Chain and importing funds to P-Chain
+		recipientOwner := secp256k1fx.OutputOwners{
+			Threshold: 1,
+			Addrs: []ids.ShortID{
+				recipientKey.Address(),
+			},
+		}
+		// Use the same outputs for both X-Chain and C-Chain exports
+		exportOutputs := []*avax.TransferableOutput{
+			{
+				Asset: avax.Asset{
+					ID: avaxAssetID,
+				},
+				Out: &secp256k1fx.TransferOutput{
+					Amt: transferAmount,
+					OutputOwners: secp256k1fx.OutputOwners{
+						Threshold: 1,
+						Addrs: []ids.ShortID{
+							keychain.Keys[0].Address(),
+						},
+					},
+				},
+			},
+		}
+
+		ginkgo.By("adding new node and waiting for it to report healthy")
+		node := e2e.AddEphemeralNode(network, testnet.FlagsMap{})
+		e2e.WaitForHealthy(node)
+
+		ginkgo.By("retrieving new node's id and pop")
+		infoClient := info.NewClient(node.GetProcessContext().URI)
+		nodeID, nodePOP, err := infoClient.GetNodeID(e2e.DefaultContext())
+		require.NoError(err)
+
+		ginkgo.By("adding the new node as a validator", func() {
+			startTime := time.Now().Add(e2e.DefaultValidatorStartTimeDiff)
+			// Validation duration doesn't actually matter to this
+			// test - it is only ensuring that adding a validator
+			// doesn't break interchain transfer.
+			endTime := startTime.Add(30 * time.Second)
+
+			rewardKey, err := factory.NewPrivateKey()
+			require.NoError(err)
+
+			const (
+				delegationPercent = 0.10 // 10%
+				delegationFee     = reward.PercentDenominator * delegationPercent
+			)
+
+			_, err = pWallet.IssueAddPermissionlessValidatorTx(
+				&txs.SubnetValidator{Validator: txs.Validator{
+					NodeID: nodeID,
+					Start:  uint64(startTime.Unix()),
+					End:    uint64(endTime.Unix()),
+					Wght:   weight,
+				}},
+				nodePOP,
+				pWallet.AVAXAssetID(),
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{rewardKey.Address()},
+				},
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{rewardKey.Address()},
+				},
+				delegationFee,
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("adding a delegator to the new node", func() {
```

Comment:
q: why do we need to make nodeID as both validator and delegator?
Also IIUC, what we are doing here is delegating some more nodeID's funds to nodeID. Why is this needed?